application/classes/Ushahidi/Repository/Form/Attribute.php

```diff
@@ -13,22 +13,80 @@
 use Ushahidi\Core\SearchData;
 use Ushahidi\Core\Entity\FormAttribute;
 use Ushahidi\Core\Entity\FormAttributeRepository;
+use Ushahidi\Core\Traits\PostValueRestrictions;
+use Ushahidi\Core\Traits\UserContext;
 
 use Ramsey\Uuid\Uuid;
 use Ramsey\Uuid\Exception\UnsatisfiedDependencyException;
 
 class Ushahidi_Repository_Form_Attribute extends Ushahidi_Repository implements
 	FormAttributeRepository
 {
+	use UserContext;
+
+	use PostValueRestrictions;
+
+	protected $form_id;
+
+	protected $form_stage_id;
+
+	protected $form_stage_repo;
+
 	// Use the JSON transcoder to encode properties
 	use Ushahidi_JsonTranscodeRepository;
 
+	public function __construct(
+			Database $db,
+			FormStageRepository $form_stage_repo,
+		)
+	{
+		parent::__construct($db);
+
+		$this->form_stage_repo = $form_stage_repo;
+
+	}
+
 	// Ushahidi_JsonTranscodeRepository
 	protected function getJsonProperties()
 	{
 		return ['options', 'config'];
 	}
 
+	protected function getFormId()
+	{
+		if (!$this->form_id && $this->form_stage_id) {
+			$form = $this->form_stage_repo->getFormByStageId($this->form_stage_id);
+			if ($form) {
+				$this->form_id = $form.id;
+			}
+		}
+	}
+
+	}
+	protected function isRestricted ()
+	{
+
+		$user = $this->getUser();
+		$this->getFormId();
+		if ($this->form_id) {
```

Comment:
Why are we maintaining the form id as state? rather than just passing it as a parameter. It seems to make things less clear..